### PR TITLE
Show brave:// instead of chrome:// in tab title for unloaded tabs

### LINKED ISSUE #53821: `chrome://<host>` is displayed on restored brave://<host> tabs when they are not loaded yet
https://github.com/user-attachments/assets/1519c949-f29d-43af-87b1-fd2939d81e0d

### PR DESCRIPTION
**Pull request by netzenbot**
## Summary
- Override `NavigationEntryImpl::GetTitleForDisplay()` to convert `chrome://` to `brave://` in the display title fallback for restored but unloaded tabs

Resolves https://github.com/brave/brave-browser/issues/53821

## Root Cause
Brave uses `chrome://` URLs internally and converts to `brave://` in the UI layer (location bar, hover cards, etc.). When a restored tab hasn't loaded yet and has no saved page title, `GetTitleForDisplay()` falls back to formatting the virtual URL. Since the virtual URL is `chrome://settings` (not `brave://settings`), the tab displays "chrome://settings" to the user.

## Fix
Override `GetTitleForDisplay()` in Brave's existing `NavigationEntryImpl` chromium_src override. When the real title is empty (fallback path), check if the cached display title starts with `chrome://` and replace with `brave://`. This catches all display paths (tab strip, window title, back/forward menu, task manager, etc.) since they all go through `GetTitleForDisplay()`.

## Test Plan
- [x] Ran `npm run format` - passed
- [x] Ran `npm run presubmit` - passed
- [x] Ran `npm run gn_check` - passed
- [x] Ran `npm run build` - passed
- [x] BraveNavigationEntryTest.GetTitleForDisplayConvertsChromeSchemeToBrave - passed
- [x] BraveNavigationEntryTest.GetTitleForDisplayPreservesExplicitTitle - passed
- [x] BraveNavigationEntryTest.GetTitleForDisplayDoesNotAffectNonChromeScheme - passed
- [ ] CI passes cleanly

### CODE REVIEW

File: chromium_src/content/browser/renderer_host/navigation_entry_impl_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  EXPECT_EQ(u"brave://flags", entry->GetTitleForDisplay());
+}
+
+TEST_F(BraveNavigationEntryTest, GetTitleForDisplayPreservesExplicitTitle) {
+  auto entry = CreateEntry(GURL("chrome://settings"));
+  entry->SetTitle(u"Settings");
+  EXPECT_EQ(u"Settings", entry->GetTitleForDisplay());
+}
+
+TEST_F(BraveNavigationEntryTest,
```
Comment (bridiver) [inline chromium_src/content/browser/renderer_host/navigation_entry_impl_unittest.cc:48]: Also add checks to this test for `brave:// scheme`, `chrome-blah`, ` chrome://settings` (empty leading space is intentional), `http://chrome.com` and `http://chrome.com/chrome://` are all unchanged

File: chromium_src/content/browser/renderer_host/navigation_entry_impl_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+       GetTitleForDisplayDoesNotAffectNonChromeScheme) {
+  auto entry = CreateEntry(GURL("https://example.com"));
+  EXPECT_EQ(u"example.com", entry->GetTitleForDisplay());
+}
+
+}  // namespace content
```
Comment (netzenbot) [inline chromium_src/content/browser/renderer_host/navigation_entry_impl_unittest.cc:54]: nit: The test constructs `GURL` objects directly but doesn't include `url/gurl.h`. Per IWYU, include `"url/gurl.h"` directly rather than relying on the transitive include from `navigation_controller.h`, which can change. https://github.com/brave/brave-core/tree/master/docs/best-practices/coding-standards.md#CS-001

**Comment by bridiver** [comment]
closed in favor of https://github.com/brave/brave-core/pull/36188
